storehaus-algebra/src/main/scala/com/twitter/storehaus/algebra/MergeableStore.scala

```diff
@@ -61,10 +66,58 @@ object MergeableStore {
       }
 
     /**
-     * Combine original keys with result after put. Some keys might have dropped,
-     * fill those using missingfn.
+     * Combine original keys with result after put.
      */
-    FutureOps.liftFutureValues(keySet, mPutResultFut, missingfn)
+    FutureOps.liftFutureValues(keySet, mPutResultFut, getFailures)
+  }
+
+  /**
+   * Collects keyed futures, partitioning out the failures.
+   */
+  private[algebra] def collectWithFailures[K, V](fs:Map[K, Future[V]]): (Future[Seq[(K, V)]], Map[K, Future[Nothing]]) = {
```

Comment:
I'm not crazy about using `Seq`. I'd rather use `List` or `Vector` so the contracts are clearer.